### PR TITLE
docs: improve torch.matrix_exp()

### LINKED ISSUE #55599: torch.matrix_exp() doc is missing a signature
See https://pytorch.org/docs/master/generated/torch.matrix_exp.html?highlight=matrix_exp#torch.matrix_exp.

It would also be nice if the mathematical expression related to the signature.

cc @brianjo @mruberry

### PR DESCRIPTION
**Pull request by neal2018**
Add a signature and make the mathematical expression related to the signature

Fixes #55599

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -5000,13 +5000,14 @@ def merge_dicts(*dicts):
 Alias for :func:`torch.linalg.matrix_power`
 """.format(**common_args))
 
-add_docstr(torch.matrix_exp,
-           r"""
+add_docstr(torch.matrix_exp, r"""
```
Comment (mruberry) [inline torch/_torch_docs.py:5003]: Thank you for submitting this PR @neal2018! While we're fixing this doc let's take a couple opportunities to improve it, too.

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -5000,13 +5000,14 @@ def merge_dicts(*dicts):
...
+matrix_exp(input) -> Tensor
+
 Returns the matrix exponential. Supports batched input.
```
Comment (mruberry) [inline torch/_torch_docs.py:5006]: "Returns the matrix exponential. Supports batched input." -> "Computes the matrix exponential of a square matrix or of each square matrix in a batch."

File: torch/_torch_docs.py [hunk 3]
```diff
@@ -5000,13 +5000,14 @@ def merge_dicts(*dicts):
...
-For a matrix ``A``, the matrix exponential is defined as
+For a matrix :attr:`input`, the matrix exponential is defined as
 
 .. math::
-    \mathrm{e}^A = \sum_{k=0}^\infty A^k / k!
+    \mathrm{e}^\text{input} = \sum_{k=0}^\infty \text{input}^k / k!
```
Comment (mruberry) [inline torch/_torch_docs.py:5010]: Nice.

**Comment by mruberry** [comment]
I just suggested one small additional tweak to the existing document. Otherwise this looks great! Just ping me when the tweak is implemented and we'll get this merged.

**Comment by neal2018** [comment]
Hey, the tweak is implemented! @mruberry

**Review by mruberry — APPROVED** [review]
Awesome! Thanks @neal2018!

**Pull request closed without merging**